yt_dlp/extractor/tiktok.py

```diff
@@ -557,21 +556,10 @@ def get_random_cookie():
             status = traverse_obj(next_data, ('props', 'pageProps', 'statusCode'), expected_type=int) or 0
             video_data = traverse_obj(next_data, ('props', 'pageProps', 'itemInfo', 'itemStruct'), expected_type=dict)
         else:
-            try:
-                sigi_data = get_element_by_attribute('id', r'(SIGI_STATE|sigi-persisted-data)', webpage, escape_value=False)
-            except compat_HTMLParseError as err:
-                self.write_debug(f'Parsing HTML failed, due to {err}')
-                sigi_data = self._search_regex(
-                    r'''(?xs)
-                        <script\s[^>]*?\b
-                            id\s*=\s*(?P<q>"|'|\b)(SIGI_STATE|sigi-persisted-data)(?P=q)[^>]*>
-                            [^=]*?=?\s*(?P<json>\{.+?\})\s*(?:;[^<]+)?
-                        </script
-                    ''', webpage, 'sigi data', group='json')
-            sigi_state = self._parse_json(sigi_data, video_id)
-
-            status = traverse_obj(sigi_state, ('VideoPage', 'statusCode'), expected_type=int) or 0
-            video_data = traverse_obj(sigi_state, ('ItemModule', video_id), expected_type=dict)
+            sigi_json = get_element_by_id('SIGI_STATE', webpage)
```

Comment:
any reason u dropped `sigi-persisted-data`?

Reply:
More than one:
It seems to be a legacy behaviour, no longer sent.
Judging from the regex it was assigned to a js variable which would require to process it before parsing as json.